RoboJackets/robocup-software
# Is done planners

## Description
Adds is_done methods to each planner, which return true if the planner is done with its job and false otherwise. This sets the groundwork for more advanced planner behavior (e.g. ball placement #1904 ) on the C++ side.

Also tuned the capture skill for ER-force simulator. Did not tune other skills as that would have taken too long.

## Associated / Resolved Issue
Resolves # or [ClickUp card](https://app.clickup.com/t/2zpw9jp)

## Design Documents
None, based the new is_done() methods off of the existing planner logic + gameplay Skill equivalents.

## Steps to test
### Test Case 1
1. Run motion planning test play named `test_motion_planning.AllBots` (second test_motion_planning play).
2. Notice the debug messages.

**Expected result:** Debug msgs say "robot x's PathTargetPlanner is_done"

Note: other planners have is_done, but are annoying to test because gameplay Skill is_dones override the planner is_dones. (Ask Kevin to provide more clarification if that doesn't make sense.)

## Key Files to Review
Gameplay Changes
 * `rj_gameplay/skill/*` - simple tweaks to existing is_dones for better performance
 * other `rj_gameplay/*` and `config/` - bugfixes or new motion planning test play

new nearly_equals()
 * `rj_common/*`
 * `rj_geometry/*`
 * `soccer/src/soccer/planning/instant.hpp`

Main Planning is_done changes
 * `soccer/src/soccer/planning/*`

## Review Checklist


*intentionally left some debug prints to illustrate working behavior, can delete after
